sdk/core/azure-core/tests/azure_core_asynctests/test_polling.py

```diff
@@ -0,0 +1,183 @@
+    """
+    def __init__(self, sleep=0):
+        self._initial_response = None
+        self._deserialization_callback = None
+        self._sleep = sleep
+
+    def initialize(self, _, initial_response, deserialization_callback):
+        self._initial_response = initial_response
+        self._deserialization_callback = deserialization_callback
+        self._finished = False
+
+    async def run(self):
+        """Empty run, no polling.
+        """
+        self._finished = True
+
+    def status(self):
+        """Return the current status as a string.
+        :rtype: str
+        """
+        return "succeeded" if self._finished else "running"
+
+    def finished(self):
+        """Is this polling finished?
+        :rtype: bool
+        """
+        return self._finished
+
+    def resource(self):
+        return self._deserialization_callback(self._initial_response)
+
+    def get_continuation_token(self):
+        return self._initial_response
+
+    @classmethod
+    def from_continuation_token(cls, continuation_token, **kwargs):
+        # type(str, Any) -> Tuple
+        initial_response = continuation_token
+        deserialization_callback = kwargs['deserialization_callback']
+        return None, initial_response, deserialization_callback
+
+
+@pytest.mark.asyncio
+async def test_poller(client):
+
+    # Same the poller itself doesn't care about the initial_response, and there is no type constraint here
+    initial_response = "Initial response"
+
+    # Same for deserialization_callback, just pass to the polling_method
+    def deserialization_callback(response):
+        assert response == initial_response
+        return "Treated: "+response
+
+    method = AsyncNoPolling()
+
+    poller = AsyncLROPoller(client, initial_response, deserialization_callback, method)
+
+    result = await poller.result()
+    assert poller.done()
+    assert result == "Treated: "+initial_response
+    assert poller.status() == "succeeded"
+
+    # Test with a basic Model
+    poller = AsyncLROPoller(client, initial_response, Model, method)
+    assert poller._polling_method._deserialization_callback == Model.deserialize
+
+    # Test poller that method do a run
+    method = PollingTwoSteps(sleep=1)
+    poller = AsyncLROPoller(client, initial_response, deserialization_callback, method)
+
+    result = await poller.result()
+    assert result == "Treated: "+initial_response
+    assert poller.status() == "succeeded"
+
+    # Test continuation token
+    cont_token = poller.continuation_token()
+
+    method = PollingTwoSteps(sleep=1)
+    new_poller = AsyncLROPoller.from_continuation_token(
+        continuation_token=cont_token,
+        client=client,
+        initial_response=initial_response,
+        deserialization_callback=Model,
+        polling_method=method
+    )
+    result = await poller.result()
+    assert result == "Treated: "+initial_response
+    assert poller.status() == "succeeded"
+
```

Comment:
Do we want to add a test for wrapping the poller in an asyncio Task to try cancellation, done_callbacks etc?
Could also do this for the other support async eventloops.

Reply:
I'll try to do that tomorrow

Reply:
https://github.com/Azure/azure-sdk-for-python/pull/10801/commits/fa71eceec10a93b652e3c31793bd38e5071fb6f2